src/middleware/insights.ts

```diff
@@ -49,6 +57,7 @@ aa('setUserToken', 'your-user-token');
     });
 
     return {
+      $$type: 'ais.insights',
```

Comment:
Why do we need `$$type` on middleware?

Reply:
I think it was needed for a previous design where InstantSearch read from the middleware array for the sendEvent function, but now it's changed so the middleware augments the InstantSearch instance directly

Reply:
Right. It's not needed any more. So I've removed it. https://github.com/algolia/instantsearch.js/pull/4456/commits/77b77a9b96f127b92f573aa0ffe77263b1408fc5